app/javascript/react/screens/App/common/ListViewToolbar/ListViewToolbar.js

```diff
@@ -201,22 +201,99 @@ class ListViewToolbar extends Component {
+        />
+        {this.renderInput()}
+      </Filter>
+    );
+  };
+
+  renderSortControls = () => {
+    const { sortFields, currentSortType, isSortNumeric, isSortAscending } = this.state;
+    return (
+      <Sort>
+        <Sort.TypeSelector
+          sortTypes={sortFields}
+          currentSortType={currentSortType}
+          onSortTypeSelected={this.updateCurrentSortType}
+        />
+        <Sort.DirectionSelector
+          isNumeric={isSortNumeric}
+          isAscending={isSortAscending}
+          onClick={this.toggleCurrentSortDirection}
+        />
+      </Sort>
+    );
+  };
+
+  renderActiveFilters = filteredSortedPaginatedListItems => {
+    const { activeFilters } = this.state;
+    return (
+      activeFilters &&
+      activeFilters.length > 0 && (
+        <Toolbar.Results>
+          <h5>
+            {filteredSortedPaginatedListItems.itemCount}{' '}
+            {filteredSortedPaginatedListItems.itemCount === 1 ? __('Result') : __('Results')}
+          </h5>
+          <Filter.ActiveLabel>{__('Active Filters')}:</Filter.ActiveLabel>
+          <Filter.List>
+            {activeFilters.map((item, index) => (
+              <Filter.Item key={index} onRemove={this.removeFilter} filterData={item}>
+                {item.label}
+              </Filter.Item>
+            ))}
+          </Filter.List>
+          <a
+            href="#"
+            onClick={e => {
+              e.preventDefault();
+              this.clearFilters();
+            }}
+          >
+            {__('Clear All Filters')}
+          </a>
+        </Toolbar.Results>
+      )
+    );
+  };
+
+  renderPaginationRow = filteredSortedPaginatedListItems => {
+    const { pagination, pageChangeValue } = this.state;
+    return (
+      <PaginationRow
+        viewType={PAGINATION_VIEW.LIST}
+        pagination={pagination}
+        pageInputValue={pageChangeValue}
+        amountOfPages={filteredSortedPaginatedListItems.amountOfPages}
+        itemCount={filteredSortedPaginatedListItems.itemCount}
+        itemsStart={filteredSortedPaginatedListItems.itemsStart}
+        itemsEnd={filteredSortedPaginatedListItems.itemsEnd}
+        onPerPageSelect={this.onPerPageSelect}
+        onFirstPage={this.onFirstPage}
+        onPreviousPage={this.onPreviousPage}
+        onPageInput={this.onPageInput}
+        onNextPage={this.onNextPage}
+        onLastPage={this.onLastPage}
+        onSubmit={this.onSubmit}
+      />
+    );
+  };
+
   render() {
-    return this.props.children(this.state, {
-      onFirstPage: this.onFirstPage,
-      onLastPage: this.onLastPage,
-      onNextPage: this.onNextPage,
-      onPageInput: this.onPageInput,
-      onPerPageSelect: this.onPerPageSelect,
-      onPreviousPage: this.onPreviousPage,
-      onSubmit: this.onSubmit,
-      clearFilters: this.clearFilters,
-      removeFilter: this.removeFilter,
-      selectFilterType: this.selectFilterType,
+    return this.props.children({
```

Comment:
I'd like to be able to keep the old API in addition to the new changes here.  This allows this component to be more reusable.  This way we can use this (after a rename perhaps) with the filter/sort/paginate controls in our tables as well

Reply:
Ok, sounds good. I had figured that up until my last commit, but had then decided to simplify it and add helpers back in as we need them. I'll just remove that last commit.